### DISCUSSION TITLE
Welcome to seed-labs Discussions!

### DISCUSSION THREAD

**Discussion by kevin-w-du** [seed-labs/seed-labs#27] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by parthian88** [comment]
trying to use docker for the first time in the seed labs, specifically the String format vulnerability one. Having an issue with the build:
[05/10/24]seed@seed-virtual-machine:~/.../Labsetup-arm$ dcbuild
/usr/local/bin/docker-compose: line 1: Not: command not found

Tried downloading via curl based on problem troubleshooting section but made no difference. Anyone have any insight with respect to this? thanks in advance.

**Reply by kevin-w-du** [reply]
For the arm, the `docker-compose` command becomes `docker compose` (no dash). You may want to modify the `dcbuild`, `dcup`, and `dcdown` aliases in the `.bashrc` file.

**Comment by parthian88** [comment]
Thank you Professor, when I executed I get:

***@***.***:~/.../Lab4$ docker compose build
WARN[0000] /home/seed/Documents/Lab4/docker-compose.yml: `version` is
obsolete
[+] Building 1.0s (11/14)
docker:default
 => [fmt-server-2 internal] load build definition from Dockerfile
0.0s
 => => transferring dockerfile: 177B
 0.0s
 => [fmt-server-1 internal] load build definition from Dockerfile
0.0s
 => => transferring dockerfile: 177B
 0.0s
 => [fmt-server-1 internal] load metadata for docker.io/handsonsecurity/s
0.9s
 => [fmt-server-1 internal] load .dockerignore
 0.0s
 => => transferring context: 2B
0.0s
 => [fmt-server-2 internal] load .dockerignore
 0.0s
 => => transferring context: 2B
0.0s
 => CANCELED [fmt-server-2 1/4] FROM docker.io/handsonsecurity/seed-ubunt
0.1s
 => => resolve ***@***.***:da9
0.0s
 => => sha256:f69a36dc31d6462eb3cfd5e29db0daa098d1fbcda65 2.81kB / 2.81kB
0.0s
 => => sha256:da995b5309b3f12181ebddda71431243ec56eaeb64c 1.36kB / 1.36kB
0.0s
 => [fmt-server-1 internal] load build context
 0.0s
 => => transferring context: 2B
0.0s
 => [fmt-server-2 internal] load build context
 0.0s
 => => transferring context: 2B
0.0s
 => ERROR [fmt-server-1 2/4] COPY server    /fmt/
0.0s
 => ERROR [fmt-server-1 3/4] COPY format-64  /fmt/format
 0.0s
 => ERROR [fmt-server-2 2/4] COPY server    /fmt/
0.0s
 => ERROR [fmt-server-2 3/4] COPY format-64  /fmt/format
 0.0s
------
 > [fmt-server-1 2/4] COPY server    /fmt/:
------
------
 > [fmt-server-1 3/4] COPY format-64  /fmt/format:
------
------
 > [fmt-server-2 2/4] COPY server    /fmt/:
------
------
 > [fmt-server-2 3/4] COPY format-64  /fmt/format:
------
failed to solve: failed to compute cache key: failed to calculate checksum
of ref 379b2c1c-888a-467d-95af-154b8fde08aa::z2ks8hprpadnij2uy06rbos7n:
"/format-64": not found

is there something else I need to adjust?

Sajjad

On Fri, May 10, 2024 at 10:31 PM Kevin Du ***@***.***> wrote:

> For the arm, the docker-compose command becomes docker compose (no dash).
> You may want to modify the dcbuild, dcup, and dcdown aliases in the
> .bashrc file.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/seed-labs/seed-labs/discussions/27#discussioncomment-9391403>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALK6UVOL3TK6B7CXJI4KSPTZBV7HPAVCNFSM4UVT6EZ2U5DIOJSWCZC7NNSXTOKENFZWG5LTONUW63SDN5WW2ZLOOQ5TSMZZGE2DAMY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Sajjad R Ibrahim

**Reply by kevin-w-du** [reply]
You need to follow the instructions in the lab description. You missed the "make install" step on page 3. This is a common mistake though.

**Comment by parthian88** [comment]
Thank you sir and my apologies. That was my mistake I missed that line.

From: Kevin Du ***@***.***>
Date: Friday, May 10, 2024 at 10:51 PM
To: seed-labs/seed-labs ***@***.***>
Cc: Sajjad R Ibrahim ***@***.***>, Comment ***@***.***>
Subject: Re: [seed-labs/seed-labs] Welcome to seed-labs Discussions! (Discussion #27)

You need to follow the instructions in the lab description. You missed the "make install" step on page 3. This is a common mistake though.
